src/codechicken/nei/recipe/GuiRecipe.java

```diff
@@ -484,9 +522,11 @@ protected void drawGuiContainerBackgroundLayer(float f, int mouseX, int mouseY)
 
         GL11.glPushMatrix();
         GL11.glTranslatef(j + 5, k + 32 + yShift, 0);
-        for (int i = page * recipesPerPage; i < handler.numRecipes() && i < (page + 1) * recipesPerPage; i++) {
-            handler.drawBackground(i);
-            GL11.glTranslatef(0, handlerInfo.getHeight(), 0);
+        try (HeightHack heightHack = new HeightHack()) {
+            for (int i = page * recipesPerPage; i < handler.numRecipes() && i < (page + 1) * recipesPerPage; i++) {
```

Comment:
What do you think about extracting  `i < handler.numRecipes() && i < (page + 1) * recipesPerPage` to some method with explanatry name, like `isInBoundsAndOnCurrentPage(recepyIndex)` (and reused on line 476)